pkg/specgenutil/specgen.go

```diff
@@ -1297,3 +1298,222 @@ func GetResources(s *specgen.SpecGenerator, c *entities.ContainerCreateOptions)
+		healthCheckOptions.Interval = *changedHealthCheckConfig.HealthStartupInterval
+		changed = true
+	}
+
+	if changedHealthCheckConfig.HealthStartupRetries != nil {
+		healthCheckOptions.Retries = int(*changedHealthCheckConfig.HealthStartupRetries)
+		changed = true
+	}
+
+	if changedHealthCheckConfig.HealthStartupTimeout != nil {
+		healthCheckOptions.Timeout = *changedHealthCheckConfig.HealthStartupTimeout
+		changed = true
+	}
+
+	if changedHealthCheckConfig.HealthStartupSuccess != nil {
+		healthCheckOptions.Successes = int(*changedHealthCheckConfig.HealthStartupSuccess)
+		changed = true
+	}
+
+	healthCheckOptions.StartPeriod = "1s"
+
+	if noHealthCheck && changed {
+		return nil, false, errors.New("cannot specify both --no-healthcheck and other HealthCheck flags")
+	}
+
+	if noHealthCheck {
+		return nil, true, nil
+	}
+
+	healthCheckConfig, err := getNewHealthCheckConfig(healthCheckOptions)
+	if err != nil {
+		return nil, false, err
+	}
+	newStartupHealthCheckConfig := new(define.StartupHealthCheck)
+	newStartupHealthCheckConfig.Schema2HealthConfig = *healthCheckConfig
+	newStartupHealthCheckConfig.Successes = healthCheckOptions.Successes
+	return newStartupHealthCheckConfig, changed, nil
+}
+
+func GetValidHealthCheckDestination(destination string) (string, error) {
+	if destination == define.HealthCheckEventsLoggerDestination || destination == define.DefaultHealthCheckLocalDestination {
+		return destination, nil
+	}
+
+	fileInfo, err := os.Stat(destination)
+	if err != nil {
+		return "", fmt.Errorf("HealthCheck Log '%s' destination error: %w", destination, err)
+	}
+	mode := fileInfo.Mode()
+	if !mode.IsDir() {
+		return "", fmt.Errorf("HealthCheck Log '%s' destination must be directory", destination)
+	}
+
+	absPath, err := filepath.Abs(destination)
+	if err != nil {
+		return "", err
+	}
+	return absPath, nil
+}
+
+func GetNewGlobalHealthCheck(changedHealthCheckConfig define.UpdateHealthCheckConfig) (define.GlobalHealthCheckOptions, error) {
+	globalOptions := define.GlobalHealthCheckOptions{}
+
+	healthLogDestination := changedHealthCheckConfig.HealthLogDestination
+	if changedHealthCheckConfig.HealthLogDestination != nil {
+		dest, err := GetValidHealthCheckDestination(*changedHealthCheckConfig.HealthLogDestination)
+		if err != nil {
+			return define.GlobalHealthCheckOptions{}, err
+		}
+		healthLogDestination = &dest
+	}
+	globalOptions.HealthLogDestination = healthLogDestination
+
+	globalOptions.HealthMaxLogSize = changedHealthCheckConfig.HealthMaxLogSize
+
+	globalOptions.HealthMaxLogCount = changedHealthCheckConfig.HealthMaxLogCount
+
+	if changedHealthCheckConfig.HealthOnFailure != nil {
+		val, err := define.ParseHealthCheckOnFailureAction(*changedHealthCheckConfig.HealthOnFailure)
+		if err != nil {
+			return globalOptions, err
+		}
+		globalOptions.HealthCheckOnFailureAction = &val
+	}
+
+	return globalOptions, nil
+}
+
+func getHealthCheckOptions(originalHealthCheckConfig interface{}) define.HealthCheckOptions {
```

Comment:
I think this can use the new IHealthCheckConfig type here instead the reflect logic in this function?
You might need to move the IHealthCheckConfig definition out of libpod for that so it can be imported on the remote client. (i.e. move it to libpod/define)